src/test/ui/function-pointer/unsized-ret.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0277]: the size for values of type `str` cannot be known at compilation time
+  --> $DIR/unsized-ret.rs:9:5
+   |
+LL |     foo::<fn() -> str, _>(None, ());
+   |     ^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
+   |
+   = help: within `fn() -> str`, the trait `Sized` is not implemented for `str`
```

Comment:
Random thought: I wonder if we could get the span for the `str` and use it for a `span_help` here, opportunistically (because for things like type aliases in a foreign crate, we might not be able to recover them).